### PR TITLE
LPD-26390 UndoRedo for locale changes

### PR DESCRIPTION
**Pull request by ambrinchaudhary**
This is a follow up of https://github.com/liferay-content-management/liferay-portal/pull/5706

I've tried to simplify things but haven't enough time to complete it. Please @NemethNorbert, review if it works, for me it does, but you have more knowledge for test this properly. If something is broken, we can go back to your pull. 

If not, I leave some comments in case you have time to analyse them.

# What is this trying to solve?

[LPD-26390](https://liferay.atlassian.net/browse/LPD-26390)
[LPD-31535](https://liferay.atlassian.net/browse/LPD-31535)
[LPD-31072](https://liferay.atlassian.net/browse/LPD-31072)

This PR is:

1. Introducing a new unique localechange event, that is fired from the language-selector. This was implemented as the original inputLocalized:localechanged event is fired in many different scenarios where a localechange actually did not happen.
2. Updated the UndoRedo state, to hold information about what language is translated in the metadatafields
3. With the translatedLanguages for the metafields we are removing and restoring translations when undoing or redoing a localizationChange
4. When the defaultLocale is changed we reset the available history steps, another subtask is opened to implement a warning for the user, that changing defaultLocale is resetting the available History steps.

# How can you verify that it works?

add to portal-ext:
`feature.flag.LPD-11228=true
feature.flag.LPD-15596=true`

-  Enable changable default language for web content
- Create a web content
- Fill the fields
- Change locale to ES
- Change the title
- Step back 2 times ( so you end up on the default locale)

**Expected**: When opening the language selector the ES locale is not translated

- Hit Redo 2 times ( so you are on the ES locale with the modified title)

**Expected** : When opening the language selector the ES locale is translated

- Change the default locale

**Expected** : You cannot use Undo or Redo as the history steps were beeing reset

### CODE REVIEW

File: modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/UndoRedo.js [hunk 1]
```diff
@@ -48,53 +48,200 @@ export default function UndoRedo({
...
+							selectedLanguageId
+						);
+					}
+				});
+			titleInputComponent.selectFlag(nextStep.selectedLanguageId);
+
+			selectedLanguageIdInput.value = nextStep.selectedLanguageId;
+		}
+
+		descriptionInputComponent.updateInputLanguage(
```
Comment (ambrinchaudhary) [inline modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/UndoRedo.js:135]: This last part is equal in both functions `handleUndo` and `handleRedo`. Could we extract it?

File: modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/UndoRedo.js [hunk 2]
```diff
@@ -109,82 +256,148 @@ export default function UndoRedo({
...
+			const selectedLanguageIdInput = document.getElementById(
+				`${portletNamespace}languageId`
 			);
 
-			const titleInputComponent = Liferay.componentReady(
-				`${portletNamespace}${META_FIELD_NAMES.title}`
-			);
+			const newHistory = {
+				defaultLanguageId: defaultLanguageIdInput.value,
+				descriptionInputComponent:
```
Comment (ambrinchaudhary) [inline modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/UndoRedo.js:265]: Maybe we could rename this to `descriptionInputValue`, so we don't confuse with the component? Same with friendlyURLInputComponent->friendlyURLInputValue and titleInputComponent-> titleInputValue

**Comment by ambrinchaudhary** [comment]
ci:test:sf

**Comment by NemethNorbert** [comment]
Hey @ambrinchaudhary,

The changes are good and I resent the PR with the requested changes here: https://github.com/liferay-content-management/liferay-portal/pull/5720

**Pull request closed without merging**